coalib/io/FileFactory.py

```diff
@@ -0,0 +1,127 @@
+import os
+
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq('name', 'timestamp')
+class FileFactory:
+    """
+    The ``FileFactory`` is a wrapper class to replace the actual file contents
+    from the ``file-dict``. The object contains the following information about
+    the file:
+
+        * The filename (absolute file path).
+        * The modification timestamp of the file.
+        * The file in the form of a string.
```

Comment:
Maybe move the "(UTF-8 decoded)" part here, the string also is decoded from the raw content, so it should be mentioned as soon as it's becoming relevant.